core/util/src/main/java/edu/cmu/cs/dennisc/math/rigidbody/TranslationAndOrientationFunction.java

```diff
@@ -207,15 +154,41 @@ protected TranslationAndOrientationDerivative evaluate(TranslationAndOrientation
   @Override
   public void update(TranslationAndOrientationDerivative a, TranslationAndOrientationDerivative b, TranslationAndOrientationDerivative c, TranslationAndOrientationDerivative d, double dt) {
     super.update(a, b, c, d, dt);
-    m_orientation.add(UnitQuaternion.createMultiplication(UnitQuaternion.createAddition(a.spin, UnitQuaternion.createAddition(UnitQuaternion.createMultiplication(UnitQuaternion.createAddition(b.spin, c.spin), 2.0), d.spin)), dt / 6));
-    m_orientation.normalize();
-    m_angularMomentum.add(Vector3.createMultiplication(Vector3.createAddition(a.torque, Vector3.createAddition(Vector3.createMultiplication(Vector3.createAddition(b.torque, c.torque), 2.0), d.torque)), dt / 6));
+    m_orientation = m_orientation.plus(a.spin.plus((b.spin.plus(c.spin)).times(2.0).plus(d.spin)).times(dt / 6)).normalized();
```

Comment:
the way my eyes crossed, i understand why you needed the old logic in this comment. can delete now tho :)

Reply:
Deleted.